kotlinlib/test/src/mill/kotlinlib/contrib/ktfmt/KtfmtModuleTests.scala

```diff
@@ -0,0 +1,150 @@
+package mill.kotlinlib.contrib.ktfmt
+
+import mainargs.Leftover
+import mill.{T, api}
+import mill.kotlinlib.KotlinModule
+import mill.main.Tasks
+import mill.testkit.{TestBaseModule, UnitTester}
+import utest.{TestSuite, Tests, assert, test}
+
+object KtfmtModuleTests extends TestSuite {
+  def tests: Tests = Tests {
+
+    val (before, after) = {
+      val root = os.Path(sys.env("MILL_TEST_RESOURCE_FOLDER")) / "contrib" / "ktfmt"
+      (root / "before", root / "after")
+    }
+
+    test("ktfmt - kotlin style") {
+      assert(
+        checkState(
+          afterFormat(before, style = "kotlin"),
+          after / "style" / "kotlin"
+        )
+      )
+    }
+
+    test("ktfmt - google style") {
+      assert(
+        checkState(
+          afterFormat(before, style = "google"),
+          after / "style" / "google"
+        )
+      )
+    }
+
+    test("ktfmt - meta style") {
+      assert(
+        checkState(
+          afterFormat(before, style = "meta"),
+          after / "style" / "meta"
+        )
+      )
+    }
+
+    test("ktfmt - dry-run") {
+      checkState(
+        afterFormat(before, format = true),
+        before
+      )
+    }
+
+    test("ktfmt - don't remove unused imports") {
+      checkState(
+        afterFormat(before, removeUnusedImports = false),
+        after / "imports"
+      )
+    }
+
+    test("ktfmt - explicit files") {
+      checkState(
+        afterFormat(before, sources = Seq("src/Example.kt")),
+        after / "style" / "kotlin"
+      )
+    }
+
+    test("formatAll") {
+
+      assert(
+        checkState(
+          afterFormatAll(before),
+          after / "style" / "kotlin"
+        )
+      )
+    }
+  }
+
+  def checkState(actualFiles: Seq[os.Path], expectedRoot: os.Path): Boolean = {
+    val expectedFiles = walkFiles(expectedRoot)
+    actualFiles.length == expectedFiles.length &&
+    actualFiles.iterator.zip(expectedFiles.iterator).forall {
+      case (actualFile, expectedFile) =>
+        val actual = os.read(actualFile)
+        val expected = os.read(expectedFile)
+        actual == expected
+    }
+  }
+
+  def afterFormat(
+      moduleRoot: os.Path,
+      style: String = "kotlin",
+      format: Boolean = true,
+      removeUnusedImports: Boolean = true,
+      sources: Seq[String] = Seq.empty
+  ): Seq[os.Path] = {
+
+    object module extends TestBaseModule with KotlinModule with KtfmtModule {
+      override def kotlinVersion: T[String] = "1.9.24"
```

Comment:
The version used here should be defined at the top of the test file. Same, for the other occurences.